### PR TITLE
Add documentation for adding a new field to an entity

### PR DESCRIPTION
**Pull request by chr-schr**
## Basics

- [X] The PR is rebased with current master
- [X] I added a line to [changelog.md](/doc/changelog.md)
- [X] Details of what I changed are in the commit messages
- [X] References to issues, e.g. `close #X`, are in the commit messages and changelog
- [ ] The buildserver is happy

## Checklist

- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## First Time Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I am using [Tailwind CSS Linting](https://tailwindcss.com/blog/introducing-linting-for-tailwindcss-intellisense)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines
- [ ] I've checked conformity to requirements
- [ ] I've checked that the requirements are tested

### CODE REVIEW

**Review by markus2330 — APPROVED** [review]
Very well done!

File: doc/backend/07adding_fields_to_models.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   This will create `up.sql` and `down.sql` inside a new migration folder under `/backend/migrations/`  
+   `up.sql` should create all necessary tables, columns, constraints as well as modify data if necessary  
+   `down.sql` should undo all changes from `up.sql`
+
+2. Run the migration  
+   `make migration`  
+   During development you can run `make migration-redo` to run `down.sql` and `up.sql` of the most recent migration
+
+3. Add new field to Entity  
+   e.g. extend the `Plants` entity in [entity.rs](https://github.com/ElektraInitiative/PermaplanT/blob/master/backend/src/model/entity.rs)
```
Comment (markus2330) [inline doc/backend/07adding_fields_to_models.md:14]: we have shortcuts which are also there to ease migration from GitHub:

```suggestion
   e.g. extend the `Plants` entity in [entity.rs](https://blob.permaplant.net/master/backend/src/model/entity.rs)
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Links should be fixed.

File: doc/backend/07adding_fields_to_models.md (hunk 1 above)
Comment (chr-schr) [reply doc/backend/07adding_fields_to_models.md]: Done

File: doc/backend/07adding_fields_to_entities.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Steps to add a field to a model
```
Comment (markus2330) [inline doc/backend/07adding_fields_to_entities.md:1]: Model is still used here.

```suggestion
# Steps to add a Field to Entities
```

File: doc/backend/07adding_fields_to_entities.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+   This will create `up.sql` and `down.sql` inside a new migration folder under `/backend/migrations/`  
+   `up.sql` should create all necessary tables, columns, constraints as well as modify data if necessary  
+   `down.sql` should undo all changes from `up.sql`
+
+2. Run the migration  
+   `make migration`  
+   During development you can run `make migration-redo` to run `down.sql` and `up.sql` of the most recent migration
+
+3. Add new field to Entity  
+   e.g. extend the `Plants` entity in [entity.rs](https://blob.permaplant.net/master/backend/src/model/entity.rs)
```
Comment (markus2330) [inline doc/backend/07adding_fields_to_entities.md:14]: Do we need to rename this folder then? Or is it different as it is related to the architecture of the backend?

**Review by markus2330 — APPROVED** [review]
LGTM, please let others review first :wink:

**Review by hatchla — APPROVED** [review]
- [ ] I've tested the code
- [ ] I've read through the whole code
- [x] I've read through the whole documentation
- [x] I've checked conformity to guidelines
- [ ] I've checked conformity to requirements
- [ ] I've checked that the requirements are tested

File: doc/backend/07adding_fields_to_entities.md (hunk 3 above)
Comment (chr-schr) [reply doc/backend/07adding_fields_to_entities.md:14]: That might be a leftover from the diesel getting started guide, where the term `model` is used - no mention of entities. I looked at a handful of other rust projects using diesel and all seem to be using the term `model`

**Pull request merged**